crates/sovereign-sdk/full-node/db/sov-db/src/rocks_db_config.rs

```diff
@@ -20,9 +23,25 @@ pub struct RocksdbConfig {
 
 impl Default for RocksdbConfig {
     fn default() -> Self {
+        let mut rlim: rlimit = rlimit {
+            rlim_cur: 0,
+            rlim_max: 0,
+        };
+        let result = unsafe { getrlimit(RLIMIT_NOFILE, &mut rlim) };
```

Comment:
Is there a way to get resource limit without unsafe code?

Reply:
Found another crate called `rlimit`. It still uses unsafe within itself, but it provides a wrapper around unsafe methods, hence we can use those methods without the use of unsafe blocks. See https://github.com/chainwayxyz/citrea/pull/811/commits/e50bf1d21da62172d34d208fed1441a0d48e3b0e